core/scripts/chaincli/handler/migrate_cron.go

```diff
@@ -0,0 +1,228 @@
+	}
+
+	proxyTx, err := proxyInstance.SetPermission(k.buildTxOpts(ctx), from, to)
+	if err != nil {
+		log.Fatalln("Error setting proxy permission", err)
+	}
+	k.waitTx(ctx, proxyTx)
+
+	log.Println("Proxy permission from", from, "to", to, "set:", helpers.ExplorerLink(k.cfg.ChainID, proxyTx.Hash()))
+
+	txReceipt, err := k.client.TransactionReceipt(ctx, proxyTx.Hash())
+	if err != nil {
+		log.Fatalln("Error getting receipt for proxy tx", err)
+	}
+	if txReceipt.Status != 1 {
+		log.Fatalln("tx", proxyTx.Hash(), "failed")
+	}
+}
+
+func (k *Keeper) registerUpkeep(ctx context.Context, name string, encryptedEmail []byte, target common.Address, gasLimit uint32, amount *big.Int) string {
+	log.Println("Registering upkeep")
+
+	registrarAddr := common.HexToAddress(k.cfg.RegistrarAddr)
+	registrarInstance, err := registrar.NewUpkeepRegistrationRequests(
+		registrarAddr,
+		k.client,
+	)
+	if err != nil {
+		log.Fatalln("Error while instantiating "+registrarAddr.String()+" to registrar", err)
+	}
+	registrarABI, err := abi.JSON(strings.NewReader(registrar.UpkeepRegistrationRequestsABI))
+	if err != nil {
+		log.Fatalln("Error generating Registrar ABI", err)
+	}
+
+	registrationData, err := registrarABI.Pack("register", name, encryptedEmail, target, gasLimit, k.fromAddr, []byte{}, amount, uint8(0))
+	if err != nil {
+		log.Fatalln("Error generating registration data", err)
+	}
+	registrationTx, err := k.linkToken.TransferAndCall(k.buildTxOpts(ctx), registrarAddr, amount, registrationData)
+	if err != nil {
+		log.Fatalln("registering", err)
+	}
+	k.waitTx(ctx, registrationTx)
+
+	log.Println("Upkeep registered:", helpers.ExplorerLink(k.cfg.ChainID, registrationTx.Hash()))
+
+	txReceipt, err := k.client.TransactionReceipt(ctx, registrationTx.Hash())
+	if err != nil {
+		log.Fatalln("Error getting receipt for upkeep register tx", err)
+	}
+	if txReceipt.Status != 1 {
+		log.Fatalln("tx", registrationTx.Hash(), "failed")
+	}
+
+	rawLog := *txReceipt.Logs[2]
+	parsedLog, err := registrarInstance.ParseLog(rawLog)
+	if err != nil {
+		log.Fatalln("Error parsing RegistrationRequested log", err)
+	}
+	registrationRequestedLog, ok := parsedLog.(*registrar.UpkeepRegistrationRequestsRegistrationRequested)
+	if !ok {
+		log.Fatalln("Error type casting RegistrationRequested log", err)
+	}
+	hash := hex.EncodeToString(registrationRequestedLog.Hash[:])
+	log.Println("Registration request hash:", hash)
+
+	return hash
+}
+
+func readCsvFile(filePath string) [][]string {
+	f, err := os.Open(filePath)
+	if err != nil {
+		log.Fatal("Unable to read input file "+filePath, err)
+	}
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	records, err := csvReader.ReadAll()
+	if err != nil {
+		log.Fatal("Unable to parse file as CSV for "+filePath, err)
+	}
+
+	return records
+}
+
+func getTargetHandler(targetFunction string) []byte {
+	targetAbi, err := abi.JSON(strings.NewReader(`[{"type":"function","inputs":[],"outputs":[],"name":"` + targetFunction + `"}]`))
```

Comment:
is `targetFunction` sensitive to things like parens? ex will `handler1` and `handler1()` produce the same results?

Reply:
Good question, it works without (), but not sure what happens with them

Reply:
It needs to be strictly the function name without parenthesis. Added an input validation for that